src/CaptureFile/CaptureFileOutputStreamTest.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+#include <gtest/gtest.h>
+#include <stdint.h>
+
+#include "CaptureFile/CaptureFileOutputStream.h"
+#include "CaptureFileConstants.h"
+#include "OrbitBase/ReadFileToString.h"
+
+namespace orbit_capture_file {
+
+TEST(CaptureFileOutputStream, Smoke) {
+  // TODO(http://b/180574275): Replace this with temporary_file once it becomes available
+  const char* temp_file_name = std::tmpnam(nullptr);
+
+  auto output_stream_or_error = CaptureFileOutputStream::Create(temp_file_name);
+  ASSERT_TRUE(output_stream_or_error.has_value()) << output_stream_or_error.error().message();
+
+  std::unique_ptr<CaptureFileOutputStream> output_stream =
+      std::move(output_stream_or_error.value());
+
+  constexpr const char* kAnswerDescription =
+      "Answer to the Ultimate Question of Life, The Universe, and Everything";
+
+  orbit_grpc_protos::ClientCaptureEvent event;
+  orbit_grpc_protos::InternedString* interned_string = event.mutable_interned_string();
+  interned_string->set_key(42);
+  interned_string->set_intern(kAnswerDescription);
+
+  output_stream->WriteCaptureEvent(event);
+  output_stream->Close();
+
+  ErrorMessageOr<std::string> file_content_or_error = orbit_base::ReadFileToString(temp_file_name);
+  ASSERT_TRUE(file_content_or_error.has_value()) << file_content_or_error.error().message();
+
+  const std::string& file_content = file_content_or_error.value();
+
+  ASSERT_GT(file_content.size(), 24);
+  ASSERT_EQ(file_content.substr(0, 4), kFileSignature);
+  uint64_t capture_section_offset = 0;
+  memcpy(&capture_section_offset, file_content.data() + 8, sizeof(capture_section_offset));
+  // Not sure if we want to check this in the test.
```

Comment:
Hmm, I would argue that it does not hurt, so we can keep it. In this case, you should drop the comment.